File: ui/backend/booking-service/src/main/java/com/tripease/booking/dto/BookingRequestDTO.java
package com.tripease.booking.dto;

import lombok.Builder;
import java.math.BigDecimal;
import java.time.LocalDate;

@Builder
public record BookingRequestDTO(
        String userId,
        Long resourceId,
        String resourceType, // "FLIGHT" or "HOTEL"
        String subType,     // "ECONOMY", "BUSINESS", "DELUXE",
        LocalDate startDate, // Check-in for Hotels and Departure of flight
        LocalDate endDate,   // Check-out for Hotels and null for flight
        int quantity,
        BigDecimal totalAmount
) {}

File: ui/frontend/src/app/core/services/booking/dto/BookingRequestDTO.ts
// booking.model.ts
export interface BookingRequestDTO {
//   userId: string; it is passed directly from api gateway as header
  resourceId: number;
  resourceType: 'FLIGHT' | 'HOTEL';
  subType: string | null;
  startDate: string; // ISO format: YYYY-MM-DD
  endDate: string | null;
  quantity: number;
  totalAmount: number;
}